lib/tapioca/dsl/helpers/graphql_type_helper.rb

```diff
@@ -0,0 +1,60 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module Dsl
+    module Helpers
+      class GraphqlTypeHelper
+        extend T::Sig
+        include Tapioca::Runtime::Reflection
+
+        sig { params(type: GraphQL::Schema::Wrapper).returns(String) }
+        def type_for(type)
+          unwrapped_type = type.unwrap
+
+          parsed_type = if unwrapped_type == GraphQL::Types::Boolean
+            "T::Boolean"
+          elsif unwrapped_type == GraphQL::Types::Float
+            qualified_name_of(Float)
+          elsif unwrapped_type == GraphQL::Types::ID
+            qualified_name_of(String)
+          elsif unwrapped_type == GraphQL::Types::Int
+            qualified_name_of(Integer)
+          elsif unwrapped_type == GraphQL::Types::ISO8601Date
+            qualified_name_of(Date)
+          elsif unwrapped_type == GraphQL::Types::ISO8601DateTime
+            qualified_name_of(DateTime)
+          elsif unwrapped_type == GraphQL::Types::JSON
+            "T::Hash[String, T.untyped]"
+          elsif unwrapped_type == GraphQL::Types::String
+            qualified_name_of(String)
+          elsif unwrapped_type <= GraphQL::Schema::Enum
+            enum_values = T.cast(unwrapped_type.enum_values, T::Array[GraphQL::Schema::EnumValue])
+            value_types = enum_values.map { |v| qualified_name_of(v.value.class) }.uniq
+            if value_types.size == 1
+              value_types.first
+            else
+              "T.any(#{value_types.join(", ")})"
+            end
+          elsif unwrapped_type <= GraphQL::Schema::InputObject
+            qualified_name_of(unwrapped_type)
+          else
```

Comment:
I am super wary of calling `==` and `<=` on random constants not under our control. I am sure you remember what happened the last time we had a `constant <= T::Enum` call 😂.

So can we instead swap this over to:
```suggestion
          parsed_type = case unwrapped_type
          when GraphQL::Types::Boolean.singleton_class
            "T::Boolean"
          when GraphQL::Types::Float.singleton_class
            qualified_name_of(Float)
          when GraphQL::Types::ID.singleton_class
            qualified_name_of(String)
          when GraphQL::Types::Int.singleton_class
            qualified_name_of(Integer)
          when GraphQL::Types::ISO8601Date.singleton_class
            qualified_name_of(Date)
          when GraphQL::Types::ISO8601DateTime.singleton_class
            qualified_name_of(DateTime)
          when GraphQL::Types::JSON.singleton_class
            "T::Hash[String, T.untyped]"
          when GraphQL::Types::String.singleton_class
            qualified_name_of(String)
          when GraphQL::Schema::Enum.singleton_class
            enum_values = T.cast(unwrapped_type.enum_values, T::Array[GraphQL::Schema::EnumValue])
            value_types = enum_values.map { |v| qualified_name_of(v.value.class) }.uniq
            if value_types.size == 1
              value_types.first
            else
              "T.any(#{value_types.join(", ")})"
            end
          when GraphQL::Schema::InputObject.singleton_class
            qualified_name_of(unwrapped_type)
          else
```